src/mongo/client/dbclient.cpp

```diff
@@ -1365,6 +1366,13 @@ void DBClientWithCommands::_findAndModify(const StringData& ns,
     commandBuilder.append("new", returnNew);
     commandBuilder.append("upsert", upsert);
 
+    if (getMaxWireVersion() >= 4) {
```

Comment:
It is not required as part of this change, and I am not sure if there is a relevant wire version to check.